src/index.js

```diff
@@ -281,17 +281,44 @@ export default function (babel) {
     return (typeof node.skinny === "boolean");
   }
 
+  function getCompletionRecords(path) {
+    let paths = [];
+
+    const add = function add(_path) {
+      if (_path) paths = paths.concat(getCompletionRecords(_path));
+    };
+
+    if (path.isIfStatement()) {
+      add(path.get("consequent"));
+      add(path.get("alternate"));
+    } else if (path.isDoExpression() || path.isFor() || path.isWhile()) {
```

Comment:
if we add a `allowLoops` param to `getCompletionRecords`, we can change this section to...

```js
} else if (path.isDoExpression()) {
  add(path.get("body"));
} else if (allowLoops && (path.isFor() || path.isWhile())) {
  add(path.get("body"));
}
```
and then completely clean up `const loop = targetPath.findParent((p) => p.isLoop() && p.getFunctionParent() === path);`, below